app/src/main/java/org/mozilla/fenix/theme/FenixColors.kt

```diff
@@ -0,0 +1,257 @@
+        textSecondary = other.textSecondary
+        textWarnings = other.textWarnings
+        accent = other.accent
+        accentBright = other.accentBright
+        accentPale = other.accentPale
+        isPrivate = other.isPrivate
+        isDark = other.isDark
+    }
+
+    /**
+     * Returns a copy of this Colors, optionally overriding some of the values.
+     */
+    fun copy(): FenixColors = FenixColors(
+        above = above,
+        scrim = scrim,
+        foundation = foundation,
+        textPrimary = textPrimary,
+        textSecondary = textSecondary,
+        textWarnings = textWarnings,
+        accent = accent,
+        accentBright = accentBright,
+        accentPale = accentPale,
+        isPrivate = isPrivate,
+        isDark = isDark
+    )
+}
+
+/**
+ * The Fenix light theme is a bright, clean, and slightly cool neutral surface.
+ * This emphasizes variable web content.
+ */
+@Suppress("TopLevelPropertyNaming")
+internal val LightColorPalette = FenixColors(
+    above = FirefoxColors.White,
+    scrim = FirefoxColors.Ink90.copy(alpha = 48f),
+    foundation = FirefoxColors.LightGrey10,
+
+    textPrimary = FirefoxColors.DarkGrey90,
+    textSecondary = FirefoxColors.DarkGrey05,
+    textWarnings = FirefoxColors.Red70,
+
+    accent = FirefoxColors.Ink20,
+    accentBright = FirefoxColors.Violet70,
+    accentPale = FirefoxColors.LightGrey30,
+
+    isPrivate = false,
+    isDark = false
+)
+
+/**
+ * The Fenix light theme is a dark, clean, and slightly cool neutral surface.
+ * This emphasizes variable web content.
+ */
+@Suppress("TopLevelPropertyNaming")
+internal val DarkColorPalette = FenixColors(
+    above = FirefoxColors.DarkGrey50,
+    scrim = FirefoxColors.Ink90.copy(alpha = 48f),
+    foundation = FirefoxColors.DarkGrey80,
+
+    textPrimary = FirefoxColors.LightGrey05,
+    textSecondary = FirefoxColors.LightGrey50,
+    textWarnings = FirefoxColors.Red30,
+
+    accent = FirefoxColors.Violet50,
+    accentBright = FirefoxColors.Violet40,
+    accentPale = FirefoxColors.DarkGrey50,
+
+    isPrivate = false,
+    isDark = true
+)
+
+/**
+ * The Fenix light theme is a bold, clean, and slightly cool neutral surface.
+ * This emphasizes variable web content.
+ */
+@Suppress("TopLevelPropertyNaming")
+internal val PrivateColorPalette = FenixColors(
+    above = FirefoxColors.Ink50,
+    scrim = FirefoxColors.Ink90.copy(alpha = 48f),
+    foundation = FirefoxColors.Ink90,
+
+    textPrimary = FirefoxColors.LightGrey05,
+    textSecondary = FirefoxColors.LightGrey60,
+    textWarnings = FirefoxColors.Red30,
+
+    accent = FirefoxColors.Violet50,
+    accentBright = FirefoxColors.Violet40,
+    accentPale = FirefoxColors.Ink50,
+
+    isPrivate = true,
+    isDark = true
+)
+
+/**
+ * A Material [Colors] implementation which sets all colors to [debugColor] to discourage usage of
+ * [MaterialTheme.colors] in preference to [FenixTheme.colors].
+ */
+internal fun debugColors(
```

Comment:
We should not break the `MaterialTheme` colors. That's what components can access and use as their defaults - but they can't access this app-specific theme here.

Reply:
Google has a sample showcasing custom themes - https://github.com/android/compose-samples/tree/main/Jetsnack which I based this work on.
There are significant differences between the small color palette available in `MaterialTheme` and what Fenix uses. In this patch I am following the palette from https://www.figma.com/file/TebIgESfWmQkMcEmwNaYZl/Android?node-id=0%3A498

Reply:
Gabriel also saw the `MaterialTheme` being too limited in https://github.com/mozilla-mobile/fenix/pull/21135#discussion_r704045614

Reply:
I used these `debugColors` just as in the sample to quickly find which components try to use `MaterialTheme` values to know to `migrate` them to our custom theme values.
Maybe we should do the same for typography?

Reply:
I am not opposed to using a custom theme. However A-C and third-party (material) libraries may use `MaterialTheme` colors as default?

Reply:
That's a great point!
Maybe we should have these themes in AC to sync theme properties and values and ensure the same UX?
Then for third-party libs .. seems like a big caveat in how everything functions..
Maybe we can try to adapt some values from our theme to `MaterialTheme` (similar to how here I am using `Color.Red` for all) and hope that small subset of colors, fonts etc will not look weird?